daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Typing.scala

```diff
@@ -370,10 +370,9 @@ private[validation] object Typing {
           checkType(paramType, KStar)
           checkType(returnType, KStar)
           introExprVar(param, paramType).checkExpr(controllers, TParties)
-          choiceObservers.foreach {
-            introExprVar(param, paramType)
-              .checkExpr(_, TParties) // FIXME #7709, be conditional on: supportsContractObservers
-          }
+          choiceObservers.foreach(
+            introExprVar(param, paramType).checkExpr(_, TParties)
```

Comment:
FYI @nickchapman-da 

That makes sense. It's not what we do on the Haskell side but since the requirements are different. In Haskell, we produce DAML-LF and want to make sure we don't accidentally introduce the wrong primitives. In Scala, we might want to run without the type checker, for instance in the scenario service, but it makes still sense to check we don't have ASTs with invalid nodes.